org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/Bean.java

```diff
@@ -132,14 +130,20 @@ public boolean equals(Object obj) {
      * @throws IllegalArgumentException
      * @throws InvocationTargetException
      */
-	private void init(Object theBean) throws NoSuchMethodException, SecurityException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
-		Field[] fields = theBean.getClass().getFields();
-		for (Field field : fields) {
+	private void init(Object theBean) throws Exception {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)